utils/swift_build_support/swift_build_support/products/swiftformat.py

```diff
@@ -0,0 +1,104 @@
+# swift_build_support/products/swiftformat.py -----------------*- python -*-
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See https://swift.org/LICENSE.txt for license information
+# See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+#
+# ----------------------------------------------------------------------------
+
+import os
+import platform
+
+from build_swift.build_swift.constants import MULTIROOT_DATA_FILE_PATH
+
+from . import cmark
+from . import foundation
+from . import libcxx
+from . import libdispatch
+from . import libicu
+from . import llbuild
+from . import llvm
+from . import product
+from . import swift
+from . import swiftpm
+from . import swiftsyntax
+from . import xctest
+from .. import shell
+
+
+class SwiftFormat(product.Product):
+    @classmethod
+    def product_source_name(cls):
+        """product_source_name() -> str
+
+        The name of the source code directory of this product.
+        """
+        return "swift-format"
+
+    @classmethod
+    def is_build_script_impl_product(cls):
+        return False
+
+    @classmethod
+    def is_swiftpm_unified_build_product(cls):
+        return True
+
+    def package_name(self):
+        return 'SwiftFormat'
+
+    def run_build_script_helper(self, action, additional_params=[]):
+        script_path = os.path.join(
+            self.source_dir, 'build-script-helper.py')
+
+        configuration = 'release' if self.is_release() else 'debug'
+
+        helper_cmd = [
+            script_path,
+            action,
+            '--toolchain', self.install_toolchain_path(),
+            '--config', configuration,
+            '--build-path', self.build_dir,
+            '--multiroot-data-file', MULTIROOT_DATA_FILE_PATH,
+            # There might have been a Package.resolved created by other builds
+            # or by the package being opened using Xcode. Discard that and
+            # reset the dependencies to be local.
+            '--update'
```

Comment:
`--update` seems suspect to run during testing when it is expected to use local sources.  @nathawes I see this is copied from the stress tester - can you explain when there would be a package.resolved created? Opening it in Xcode in particular would create a resolved file in a location used by Xcode, not where it is for command-line builds.

Reply:
I'm not sure off of the top of my head, but we [tell contributors](https://github.com/apple/swift-stress-tester/blob/master/README.md#for-local-development) they can invoke build-script-helper themselves passing a swift.org toolchain path in addition to using it to to get swiftpm to generate an Xcode project the old way (we add additional search paths via an .xcconfig file), so my guess would be one of those does it. Looks like Alex added this in https://github.com/apple/swift/pull/28005. @ahoppen is that what was creating the problematic package.resolved here?

Reply:
For what it's worth, I'm okay with taking this change without resolving this issue, since it's evidently not causing problems for the stress tester.  But we should figure it out.

Reply:
The issue is what’s described in the comment above it 😉. Maybe a little more context. 
1. You build `swift-format` using this build script, `Package.resolved` gets set to the local dependencies.
2. You open the project in Xcode. `SWIFTCI_USE_LOCAL_DEPS` is not set and thus `Package.resolved` gets reset to use remote dependencies.
3. You run a build using this script again. Without the `--update` option (which again re-generates `Package.resolved`), we are now trying to create a unified build of a local SwiftSyntax and a swift-format that takes a remote dependency. 

I can’t recall the exact error scenario, but possible options include:
- SwiftPM isn’t happy if the versions of SwiftSyntax don’t refer to the same source code
- I wanted to make sure that we are definitely using the local dependency and are never stuck on a remote dependency.

Reply:
Hmm, Xcode will use its own Package.resolved that would not interfere with the local build.  However, I suppose if you built from command-line using `swift build` it would do the same thing you are describing.

Reply:
@benlangmuir Does it? I’m fairly sure, Xcode uses the same `Package.resolved`. Otherwise there also wouldn’t be any point in checking `Package.resolved` into source control (for other projects).